packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1966,27 +1966,15 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     return enabledColor;
   }
 
-  Color _getFillColor(ThemeData themeData) {
+  Color _getFillColor(ThemeData themeData, InputDecorationTheme defaults) {
     if (decoration.filled != true) { // filled == null same as filled == false
       return Colors.transparent;
     }
     if (decoration.fillColor != null) {
       return MaterialStateProperty.resolveAs(decoration.fillColor!, materialState);
     }
 
-    // dark theme: 10% white (enabled), 5% white (disabled)
-    // light theme: 4% black (enabled), 2% black (disabled)
-    const Color darkEnabled = Color(0x1AFFFFFF);
-    const Color darkDisabled = Color(0x0DFFFFFF);
-    const Color lightEnabled = Color(0x0A000000);
-    const Color lightDisabled = Color(0x05000000);
-
-    switch (themeData.brightness) {
-      case Brightness.dark:
-        return decoration.enabled ? darkEnabled : darkDisabled;
-      case Brightness.light:
-        return decoration.enabled ? lightEnabled : lightDisabled;
-    }
+    return decoration.enabled ? defaults.fillColor!: defaults.disabledFillColor!;
```

Comment:
Could the M2 default theme return a MaterialStateColor here?

Reply:
Done